File: dcp-solutions/2024-12-15/solution.rb
```ruby
def permute(nums)
  if nums.size == 1
    [nums]
  else
    result = []
    (0...nums.size).each do |i|
      num = nums[i]
      rest = nums[i+1..-1].clone
      perms = permute(rest)
      perms.each do |perm|
        result << [num, *perm]
      end
    end
    result
  end
end
```

File: dcp-solutions/2024-12-15/solution.php
```php
<?php
function permute($nums) {
    if (count($nums) == 1) {
        return [$nums];
    }

    $result = [];
    for ($i = 0; $i < count($nums); $i++) {
        $currentNum = $nums[$i];
        $restNumbers = array_merge(array_slice($nums, 0, $i), array_slice($nums, $i + 1));
        foreach (permute($restNumbers) as $permutation) {
            $result[] = array_merge([$currentNum], $permutation);
        }
    }

    return $result;
}
```